core/network/fabric-interop-cc/libs/utils/utils.go

```diff
@@ -337,86 +387,117 @@ func ReclaimAsset(ctx contractapi.TransactionContextInterface, assetType, id, re
 }
 
 // GetAssetPledgeStatus returns the asset pledge status.
-func GetAssetPledgeStatus(ctx contractapi.TransactionContextInterface, assetType, id, recipientNetworkId, recipientCert string, blankAssetJSON []byte) ([]byte, string, string, error) {
+func GetAssetPledgeStatus(ctx contractapi.TransactionContextInterface, pledgeId, recipientNetworkId, recipientCert string, blankAssetJSON []byte) ([]byte, string, string, error) {
 	// (Optional) Ensure that this function is being called by the relay via the Fabric Interop CC
 
-	pledge := &AssetPledge{
+	pledge := &common.AssetPledge{
 		AssetDetails: blankAssetJSON,
 		LocalNetworkID: "",
 		RemoteNetworkID: "",
-		RecipientCert: "",
+		Recipient: "",
 		ExpiryTimeSecs: 0,
 	}
-	pledgeJSON, err := json.Marshal(pledge)
+	pledgeBytes64, err := marshalAssetPledge(pledge)
 	if err != nil {
 		return nil, "", "", err
 	}
 
-	pledgeKey := getAssetPledgeKey(assetType, id)
-	lookupPledgeJSON, err := ctx.GetStub().GetState(pledgeKey)
+	pledgeKey := getAssetPledgeKey(pledgeId)
+	lookupPledgeBytes, err := ctx.GetStub().GetState(pledgeKey)
 	if err != nil {
-		return nil, string(pledgeJSON), string(pledgeJSON), fmt.Errorf("failed to read asset pledge status from world state: %v", err)
+		return nil, pledgeBytes64, pledgeBytes64, fmt.Errorf("failed to read asset pledge status from world state: %v", err)
 	}
-	if lookupPledgeJSON == nil {
-		return nil, string(pledgeJSON), string(pledgeJSON), nil      // Return blank
+	if lookupPledgeBytes == nil {
+		return nil, pledgeBytes64, pledgeBytes64, nil      // Return blank
 	}
-	var lookupPledge AssetPledge
-	err = json.Unmarshal(lookupPledgeJSON, &lookupPledge)
+	lookupPledge := &common.AssetPledge{}
+	err = proto.Unmarshal(lookupPledgeBytes, lookupPledge)
 	if err != nil {
-		return nil, string(pledgeJSON), string(pledgeJSON), err
+		return nil, pledgeBytes64, pledgeBytes64, err
 	}
 
 	// Match pledge with request parameters
-	if lookupPledge.RecipientCert != recipientCert || lookupPledge.RemoteNetworkID != recipientNetworkId {
-		return nil, string(pledgeJSON), string(pledgeJSON), nil      // Return blank
+	if lookupPledge.Recipient != recipientCert || lookupPledge.RemoteNetworkID != recipientNetworkId {
+		return nil, pledgeBytes64, pledgeBytes64, nil      // Return blank
+	}
+	
+	lookupPledgeBytes64, err := marshalAssetPledge(lookupPledge)
+	if err != nil {
+		return nil, pledgeBytes64, pledgeBytes64, err
+	}
+	
+	return lookupPledge.AssetDetails, lookupPledgeBytes64, pledgeBytes64, nil
+}
+
+// GetAssetPledgeDetails returns the asset pledge details for local network.
+func GetAssetPledgeDetails(ctx contractapi.TransactionContextInterface, pledgeId string) ([]byte, string, error) {
```

Comment:
Isn't this function redundant, given we have `GetAssetPledgeStatus` above?

Reply:
`GetAssetPledgeStatus`, I saw this function has lots of safety checks required during Asset Transfer protocol, which requires RecipientCert, RemoteNetworkID etc.

`GetAssetPledgeDetails`, is a simple query function for local network, to fetch all the details using just the `pledgeId`.

Reply:
Got it.